uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js

```diff
@@ -187,22 +186,35 @@ define(['controllers/controllers',
 
                     return result;
 
-                }).then(function(result){
+                }).then(function (result) {
 
                     $scope.document.id = result.document.id;
                     $scope.document.case_id = result.document.case_id;
                     $scope.document.file_count = $scope.files.length + uploader.queue.length;
 
+                    $scope.document.open = $scope.openNew;
+
                     AssignmentService.updateTab();
                     $modalInstance.close($scope.document);
                     $scope.$broadcast('ta-spinner-hide');
-                },function(reason){
+                }, function (reason) {
                     CRM.alert(reason, 'Error', 'error');
                     $modalInstance.dismiss();
                     $scope.$broadcast('ta-spinner-hide');
                     return $q.reject();
                 });
-            }
+            };
+
+            $scope.dpOpen = function ($event, name) {
+                $event.preventDefault();
+                $event.stopPropagation();
+                $scope.dpOpened[name] = true;
+            };
 
+            $scope.dropzoneClick = function () {
+                $timeout(function () {
+                    document.getElementById($rootScope.prefix + 'document-files').click();
```

Comment:
I understand the time constraints, but this is a design flaw. AngularJS controllers should never manipulate the DOM directly, that's what the directives (through the `link` function) are for